fetch/origin/no-cors.window.js

```diff
@@ -76,3 +74,99 @@ promise_test(async function () {
   assert_equals(json[0], "null");
   assert_equals(json[1], "null");
 }, "Origin header and POST navigation");
+
+var NavigationDestination = {
+  SameOrigin: 1,
+  CrossOrigin: 2,
+  properties: {
+    1: {name: "Same-Origin", origin: origins.HTTP_ORIGIN},
+    2: {name: "Cross-Origin", origin: origins.HTTP_REMOTE_ORIGIN},
+  }
+}
+
+function referrerPolicyFunctor(referrerPolicy, navigationDestination, expectedOrigin) {
+  return async function () {
+    const stash = token(),
+          referrerPolicyPath = "/fetch/origin/resources/referrer-policy.py";
+          redirectPath = "/fetch/origin/resources/redirect-and-stash.py";
+
+
+    let url = NavigationDestination.properties[navigationDestination].origin +
+              redirectPath + "?stash=" + stash;
+
+    await new Promise(resolve => {
+      const frame = document.createElement("iframe");
+      frame.src = origins.HTTP_ORIGIN + referrerPolicyPath +
+                  "?referrerPolicy=" + referrerPolicy;
+      self.addEventListener("message", e => {
+        if (e.data === "loaded") {
+          resolve();
+          frame.remove();
+        }
+      }, { once: true });
+      frame.onload = () => {
```

Comment:
This should try to avoid the onload event that can occur from the initial about:blank.

Reply:
@jdm  I try to steal some ideas in current wpt, but almost every wpt with `frame.src` set comes a `frame.onload`.

Reply:
Yeah, I'm not getting two load events here so I think this is fine. @jdm or are you saying there's a race condition with that? (I think there are some scenarios where it can occur, but not racily, though I cannot recall exactly how.)

Reply:
IIRC different browsers suppress or do not suppress the about:blank onload event if another navigation has occurred.

Reply:
let's use message-based